platform/darwin/src/NSBundle+MGLAdditions.m

```diff
@@ -6,12 +6,19 @@ @implementation NSBundle (MGLAdditions)
 
 + (instancetype)mgl_frameworkBundle {
     NSBundle *bundle = [self bundleForClass:[MGLAccountManager class]];
-    if (![bundle.infoDictionary[@"CFBundlePackageType"] isEqualToString:@"FMWK"] && !bundle.mgl_resourcesDirectory) {
+
+    if (![bundle.infoDictionary[@"CFBundlePackageType"] isEqualToString:@"FMWK"]) {
         // For static frameworks, the bundle is the containing application
-        // bundle but the resources are still in the framework bundle.
-        bundle = [NSBundle bundleWithPath:[bundle.privateFrameworksPath
-                                           stringByAppendingPathComponent:@"Mapbox.framework"]];
+        // bundle but the resources are in Mapbox.bundle.
+        NSString *bundlePath = [bundle pathForResource:@"Mapbox" ofType:@"bundle"];
+        if (bundlePath) {
+            bundle = [self bundleWithPath:bundlePath];
+        } else {
+            [NSException raise:@"Bundle not found" format:
```

Comment:
Per #7258, the exception name should be camel-cased and prefixed with the class prefix.